Return-Path: <thecivvie@tcob1.net>
Received: by tcob1.net (Postfix, from userid 505)
	id BC27C32E5F; Thu,  4 Jan 2001 21:15:09 +0000 (GMT)
Received: from localhost (tcob1.net [127.0.0.2])
	by tcob1.net (Postfix) with ESMTP id 93BE632DCF
	for <thecivvie@tcob1.net>; Thu,  4 Jan 2001 19:56:29 +0000 (GMT)
Delivered-To: thecivvie@softhome.net
Received: from pop.softhome.net [204.144.231.20]
	by localhost with POP3 (fetchmail-5.6.1)
	for thecivvie@tcob1.net (single-drop); Thu, 04 Jan 2001 19:56:29 +0000 (GMT)
Received: (qmail 21668 invoked by uid 417); 4 Jan 2001 19:09:18 -0000
Received: from 24-216-111-171.hsacorp.net (HELO easysw.com) (24.216.111.171)
  by mx0c.softhome.net with SMTP; 4 Jan 2001 19:09:18 -0000
Received: from easysw.com (bravo.easysw.com [199.10.112.130])
	by easysw.com (8.10.1/8.10.2) with ESMTP id f04J6NA24667;
	Thu, 4 Jan 2001 14:06:33 -0500
Sender: mike@easysw.com
Message-ID: <3A54C9AA.ED1ADF4A@easysw.com>
Date: Thu, 04 Jan 2001 14:06:18 -0500
From: Michael Sweet <mike@easysw.com>
Organization: Easy Software Products
X-Mailer: Mozilla 4.7C-SGI [en] (X11; I; IRIX 6.5 IP22)
X-Accept-Language: en, fr
MIME-Version: 1.0
To: thecivvie@softhome.net
Subject: Patch for vacation program...
Content-Type: multipart/mixed;
 boundary="------------36AB7C1CAB733F55D7A05395"
Resent-From: Sean Rima <thecivvie@tcob1.net>
Resent-Date: Thu, 4 Jan 2001 21:15:09 +0000
Resent-To: sean.rima@tcob1.net
Resent-Message-Id: <20010104211509.BC27C32E5F@tcob1.net>
Lines: 76

This is a multi-part message in MIME format.
--------------36AB7C1CAB733F55D7A05395
Content-Type: text/plain; charset=us-ascii
Content-Transfer-Encoding: 7bit

Sean,

Thanks for making your vacation program available.

I've run into one small problem, and have included a patch to
fix it.  Basically, if the vacation program is run more than
once at the same time (a pretty common thing with sendmail)
then the gdbm_open() call will fail with EAGAIN (because the
database is opened in exclusive mode).

A simple wrapper around the open call to retry at random intervals
fixes the problem.  Patch attached.

-- 
______________________________________________________________________
Michael Sweet, Easy Software Products                  mike@easysw.com
Printing Software for UNIX                       http://www.easysw.com
--------------36AB7C1CAB733F55D7A05395
Content-Type: text/plain; charset=us-ascii;
 name="vacation.patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline;
 filename="vacation.patch"

*** vacation.c.orig	Thu Jan  4 14:00:49 2001
--- vacation.c	Thu Jan  4 14:00:49 2001
***************
*** 158,166 ****
  	   "vacation: no such directory %s.\n", pw->pw_dir);
      exit(1);
    }
!   
!   db = gdbm_open(VDB, 128, ((iflag || nflag) ? GDBM_NEWDB : GDBM_WRITER),
! 		 0644, NULL);
    if (!db) {
      syslog(LOG_NOTICE, "vacation: %s: %s\n", VDB, strerror(errno));
      exit(1);
--- 158,173 ----
  	   "vacation: no such directory %s.\n", pw->pw_dir);
      exit(1);
    }
! 
!   do
!   {
!     db = gdbm_open(VDB, 128, ((iflag || nflag) ? GDBM_NEWDB : GDBM_WRITER),
! 		   0644, NULL);
!     if (!db && errno == EAGAIN)
!       sleep((rand() & 7) + 1);
!   }
!   while (!db && errno == EAGAIN);
!       
    if (!db) {
      syslog(LOG_NOTICE, "vacation: %s: %s\n", VDB, strerror(errno));
      exit(1);

--------------36AB7C1CAB733F55D7A05395--



